pkg/sink/sink_test.go

```diff
@@ -231,6 +231,189 @@ func TestHandleEvent(t *testing.T) {
 	}
 }
 
+func TestHandleEventWithInterceptors(t *testing.T) {
```

Comment:
This is repeating `TestHandleEvent` with only a few differences. We could use a testing table so that we could test [0-2] interceptors in a bit less verbose way. This seems fine though.

Reply:
Yeah long term, I want to extract the common bits out into a couple of helper methods that create the required resources, and the clients (like we do in the reconciler tests in pipeline). For this PR though, I wanted to keep the non-fix changes to a minimum